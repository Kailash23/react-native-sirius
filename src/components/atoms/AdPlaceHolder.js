import React, { useState } from 'react';
import { ENVIRONMENT } from '@rootConfig/apiConfig';
import { Viewport } from '@skele/components';
import { View, Text } from 'react-native';
import { withTheme } from 'react-native-paper';
import { black, white } from '@styles/colors';
import {
  SIZES,
  AD_HEIGHT,
  TABLET_AD_HEIGHT,
  PADDING_BW_ITEMS
} from '@styles/sizes';
import { customFonts } from '@styles/customFonts';
import { localStrings } from '@utils/i18n';
import { PublisherBanner } from 'react-native-admob';
import { withNavigation } from '@react-navigation/compat';
import { adProperty } from '@rootConfig/build.config';
import { layout } from '@utils/Layout';
//TODO: Transport to context
function AdPlaceHolder(props) {
  const { theme, style } = props;
  const [adLoaded, setAdLoaded] = useState(false);
  const [isInViewPort, setIsInViewPort] = useState(false);
  let adsContext = props.adsContext;
  const adSize =
    layout.isMobile || props.isArticle ? 'mediumRectangle' : 'leaderboard';
  const adStyle =
    layout.isMobile || props.isArticle
      ? styles.mainContainer
      : styles.mainTabletContainer;
  const environment = ENVIRONMENT == 'dev' ? 'staging' : 'prod';
  const ViewPortAwareView = Viewport.Aware(View);
  if (adsContext === '/home') {
    adsContext = '/homepage';
  }
  const assetTags = props.assetTags ? props.assetTags : null;
  const storyuuid = props.storyuuid ? props.storyuuid : null;
  const customTargeting = {
    cutpoint: ['small', 'app'],
    environment: [environment]
  };
  if (assetTags) {
    customTargeting.kvng = assetTags;
  }
  if (storyuuid) {
    customTargeting.assetid = [storyuuid];
  }
  const showAd = () => {
    return isInViewPort ? (
      <PublisherBanner
        adSize={adSize}
        validAdSizes={[adSize]}
        adUnitID={`${adProperty.slice(0, -1)}${adsContext}`}
        testDevices={[PublisherBanner.simulatorId]}
        customTargeting={customTargeting}
        didFailToReceiveAdWithError={e =>
          console.log('Banner:onAdFailedToLoad', e)
        }
        onAdLoaded={() => setAdLoaded(true)}
        onAdFailedToLoad={e => console.log('Banner:onAdFailedToLoad', e)}
        onAdOpened={() => console.log('Banner:onAdOpened')}
        onAdClosed={() => console.log('Banner:onAdClosed')}
        onAdLeftApplication={() => console.log('Banner:onAdLeftApplication')}
        onAppEvent={event => console.log('Banner:onAppEvent', event)}
      />
    ) : (
        false
      );
  };

  return (
    <View
      style={[
        style,
        adLoaded ? adStyle : styles.mainContainerCollapsed,
        {
          borderColor: theme.dark ? black : theme.colors.disabled,
          backgroundColor: theme.colors.adBox,
          marginTop: SIZES.PADDING
        }
      ]}>
      <ViewPortAwareView
        retainOnceInViewport={true}
        onViewportEnter={() => setIsInViewPort(true)}
        onViewportLeave={() => console.log('Ad exit')}>
        <Text style={[styles.adTitle, { color: theme.colors.text }]}>
          {props.isArticle
            ? localStrings('articleContinues')
            : localStrings('ad')}
        </Text>
        {showAd()}
      </ViewPortAwareView>
    </View>
  );
}

const styles = {
  mainContainer: {
    flex: 1,
    height: AD_HEIGHT,
    borderWidth: 0.5,
    alignContent: 'center',
    alignItems: 'center',
    justifyContent: 'center'
  },
  mainTabletContainer: {
    flex: 1,
    height: TABLET_AD_HEIGHT,
    borderWidth: 0.5,
    alignContent: 'center',
    alignItems: 'center',
    justifyContent: 'center'
  },
  mainContainerCollapsed: {
    display: 'none'
  },
  adContainer: {
    height: '75%',
    marginVertical: 10,
    marginHorizontal: '12%'
  },
  adTitle: {
    fontSize: SIZES.MINIFONT,
    marginTop: -10,
    marginBottom: 5,
    textAlign: 'center',
    fontFamily: customFonts.merriweatherSansRegular
  }
};

export default withTheme(withNavigation(AdPlaceHolder));
