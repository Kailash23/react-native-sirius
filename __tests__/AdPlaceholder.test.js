import AdPlaceHolder from '@atoms/AdPlaceHolder';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

configure({ adapter: new Adapter() });

describe('Testing <AdPlaceHolder/>', () => {
  test('should render a AdPlaceHolder', () => {
    const wrapper = shallow(<AdPlaceHolder />);
    expect(wrapper).toMatchSnapshot();
  });
});
