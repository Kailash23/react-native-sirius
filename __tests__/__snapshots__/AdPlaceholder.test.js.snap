// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <AdPlaceHolder/> should render a AdPlaceHolder 1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
/>
`;
